### PR TITLE
Truncate title of the error message to 100 characters (#46)

### LINKED ISSUE #46: Truncate issue titles
They are super long! https://github.com/atom/tree-view/issues/399

### PR DESCRIPTION
**Pull request by jportela**
Closes #46.

Creates a TitleUtilities (it may be too specific, but we do have to share the title format code for NotificationElement and NotificationIssue somewhere).

The truncation is done for 100 characters, so the github issues shouldn't span more than 2 lines.

Added a spec to verify it's truncating the notification title, unfortunately I couldn't add a spec to verify it's also truncating the create issue query string (it always shows the "a is not defined" error as a message on the link, for some reason...)

Let me know if this is completely wrong, this is my first pull request, and I'm happy to work this out :)

### CODE REVIEW

**Comment by lee-dohm** [comment]
Since the notifications library is used for more than just posting errors on GitHub, I decided to check to ensure that your changes didn't break displaying other notifications. It appears that your change affects all notifications though:

screen shot 2015-05-02 at 11 40 04 am

I used `atom.notifications.addInfo("Scopes at Cursor\n\n1234567890\n1234567890\n1234567890\n1234567890\n1234567890\n1234567890\n1234567890\n1234567890\n1234567890\n1234567890\n", {dismissable: true})` in the Developer Console to display the above notification. You'll notice if you switch back to `master` that the truncation does not occur:

screen shot 2015-05-02 at 11 45 06 am

**Comment by jportela** [comment]
Thanks for pointing that, I think I'll just truncate the url query string that is used to create the issue, which is a much more targeted change without other side effects

**Comment by lee-dohm** [comment]
:+1: Sounds like a good plan :grinning:

**Comment by jportela** [comment]
Implemented this as advised, only on the NotificationIssue class (getIssueTitle). From what I've seen, this class is only used for searching and creating issues, so I only changed the truncation to the getIssueTitle method, in order for search/creation to be consistent (I assumed that once we have truncated titles, we want to also search by the truncated title, let me know if I'm not correct). 

I've also added a spec for the Create Issues button URL.

File: lib/notification-issue.coffee [hunk 1]
```diff
@@ -52,7 +54,10 @@ class NotificationIssue
       "#{repoUrl}/issues/new?title=#{@encodeURI(@getIssueTitle())}&body=#{@encodeURI(issueBody)}"
 
   getIssueTitle: ->
-    @notification.getMessage()
+    title = @notification.getMessage()
+    if title.length > TRUNCATE_TITLE_TO
```
Comment (lee-dohm) [inline lib/notification-issue.coffee]: It looks like checking the length isn't necessary:

```javascript
foo = '123'    // => "123"
foo = foo.substring(0, 10) + '...'     // => "123..."
```

So 58-60 can just be compressed into:

```coffee
title.substring(0, TRUNCATE_TITLE_TO - 3) + '...'
```

Comment (jportela) [inline lib/notification-issue.coffee]: We only want to add the ... ellipsis if the length is bigger than 100 characters, right?

Comment (lee-dohm) [inline lib/notification-issue.coffee]: Ah ... good point. That's what I get for code reviewing before I've had my breakfast :laughing:

**Pull request closed without merging**